### PR TITLE
[Gym] Add AIME 

### PR DESCRIPTION
**Pull request by lynnliu030**

### CODE REVIEW

File: skyrl-train/examples/algorithms/dapo/run_dapo_aime.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  trainer.eval_batch_size=1024 \
+  trainer.eval_before_train=false \
+  trainer.eval_interval=5 \
+  trainer.update_epochs_per_batch=1 \
+  trainer.train_batch_size=1024 \
+  trainer.policy_mini_batch_size=256 \
+  trainer.micro_forward_batch_size_per_gpu=64 \
+  trainer.micro_train_batch_size_per_gpu=64 \
+  trainer.ckpt_interval=10 \
+  trainer.max_prompt_length=512 \
```
Comment (SumanthRH) [inline skyrl-train/examples/algorithms/dapo/run_dapo_aime.sh]: Let's use the same settings as DAPO did? 

https://github.com/volcengine/verl/blob/bd756c15c8873e40cfbb5ef2a4d1d6cadb4ea272/recipe/dapo/run_dapo_qwen2.5_32b.sh#L17-L18

Comment (SumanthRH) [reply skyrl-train/examples/algorithms/dapo/run_dapo_aime.sh]: I think the only difference here is like batch size and model arch, but beyond that we should match all other settings

File: skyrl-train/examples/algorithms/dapo/run_dapo_aime.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+OVERLONG_BUFFER_LEN=512
+OVERLONG_BUFFER_PENALTY_FACTOR=1.0
+
+# other DAPO parameters
+USE_KL_LOSS=false
+TEMPERATURE=1.0
+TOP_P=1.0
+EVAL_TOP_P=0.7
+CLIP_RATIO_C=10.0
+MAX_RESPONSE_LENGTH=1024
```
Comment (SumanthRH) [inline skyrl-train/examples/algorithms/dapo/run_dapo_aime.sh]: This one as well: https://github.com/volcengine/verl/blob/bd756c15c8873e40cfbb5ef2a4d1d6cadb4ea272/recipe/dapo/run_dapo_qwen2.5_32b.sh#L17-L18

**Review by SumanthRH — COMMENTED** [review]
Looks good overall!

For a sanity check on the reward: Could we run evaluation loop before training once for a base model on AIME and makes sure the numbers look fine? maybe Qwen3-8B or [DeepSeek-R1-0528-Qwen3-8B](https://huggingface.co/deepseek-ai/DeepSeek-R1-0528-Qwen3-8B) ?

File: skyrl-train/examples/algorithms/dapo/run_dapo_aime.sh (hunk 1 above)
Comment (lynnliu030) [reply skyrl-train/examples/algorithms/dapo/run_dapo_aime.sh]: Sounds good, will match their setup

File: skyrl-gym/skyrl_gym/envs/aime/utils.py [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    """
+    if strict_box_verify:
+        correct, pred = is_correct_strict_box(solution_str, answer, pause_tokens_index)
+        return correct == 1, pred
+
+    correct, pred = is_correct_minerva(solution_str, answer)
+    return correct, pred
+
+
+def compute_score(
```
Comment (SumanthRH) [inline skyrl-gym/skyrl_gym/envs/aime/utils.py:246]: Could we also have a few tests with typical corner cases for this? can be examples straight from aime

Comment (lynnliu030) [reply skyrl-gym/skyrl_gym/envs/aime/utils.py:246]: Added!

File: skyrl-gym/skyrl_gym/envs/aime/env.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        assert "reward_model" in extras, "reward_model field is required"
+        assert "ground_truth" in extras["reward_model"], "ground_truth is required in reward_model field"
+        self.ground_truth = extras["reward_model"]["ground_truth"]
+
+    def step(self, action: str) -> BaseTextEnvStepOutput:
+        done = True  # always done after one step
+
+        # NOTE(shu): manually parse the action with end_string |im_end|
+        # Might want the env to have a tokenizer and being able to take tokens for special reward calculation
+        action = action.split("<|im_end|>")[0]
```
Comment (SumanthRH) [inline skyrl-gym/skyrl_gym/envs/aime/env.py]: is this needed? this EOS token will be model specific ( this will work only for qwen then). Can probably make our parsing be more robust

File: skyrl-gym/tests/test_aime.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import skyrl_gym
+import pytest
+from omegaconf import DictConfig
+
+
+@pytest.mark.parametrize(
```
Comment (SumanthRH) [inline skyrl-gym/tests/test_aime.py:6]: Another point on eos token: we should also test for different eos tokens here

**Review by SumanthRH — COMMENTED** [review]
Thanks! Would be good to remove hardcoded eos token rn, and also incorporate eos token in tests

File: skyrl-gym/skyrl_gym/envs/aime/env.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+from skyrl_gym.envs.base_text_env import BaseTextEnv, BaseTextEnvStepOutput
+from skyrl_gym.envs.aime import utils
+from typing import Dict, Any
+from omegaconf import DictConfig
+
+
+class AIMEEnv(BaseTextEnv):
+    """
+    Environment for Math execution tasks.
```
Comment (SumanthRH) [inline skyrl-gym/skyrl_gym/envs/aime/env.py:9]: Add a disclaimer on the parsing here as well please, since we use `<` now to detect eos token.

**Review by SumanthRH — APPROVED** [review]
Thanks! Before merging, it would be great to add the eval accuracy we get with the AIME env for a 8B or smaller model rn and compare with reported
